keras_cv/layers/preprocessing/aug_mix.py

```diff
@@ -0,0 +1,310 @@
+        self.auto_contrast = layers.AutoContrast(value_range=self.value_range)
+        self.equalize = layers.Equalization(value_range=self.value_range)
+
+    @staticmethod
+    def _sample_from_dirichlet(alpha):
+        gamma_sample = tf.random.gamma(shape=(), alpha=alpha)
+        return gamma_sample / tf.reduce_sum(gamma_sample, axis=-1, keepdims=True)
+
+    @staticmethod
+    def _sample_from_beta(alpha, beta):
+        sample_alpha = tf.random.gamma((), 1.0, beta=alpha)
+        sample_beta = tf.random.gamma((), 1.0, beta=beta)
+        return sample_alpha / (sample_alpha + sample_beta)
+
+    def _sample_level(self, level, maxval, dtype):
+        level = (
+            self._random_generator.random_uniform(
+                shape=(), minval=0.01, maxval=level, dtype=tf.float32
+            )
+            * 10
+        )
+
+        return tf.cast((level) * maxval / 10.0, dtype)
+
+    def _loop_on_depth(self, depth_level, image_aug):
+        op_index = self._random_generator.random_uniform(
+            shape=(), minval=0, maxval=8, dtype=tf.int32
+        )
+        image_aug = self._apply_op(image_aug, op_index)
+        depth_level += 1
+        return depth_level, image_aug
+
+    def _loop_on_width(self, image, chain_mixing_weights, curr_chain, result):
+        image_aug = tf.identity(image)
+        depth = tf.cond(
+            tf.greater(self.depth, 0),
+            lambda: self.depth,
+            lambda: self._random_generator.random_uniform(
+                shape=(), minval=1, maxval=3, dtype=tf.int32
+            ),
+        )
+
+        depth_level = tf.constant([0], dtype=tf.int32)
+        depth_level, image_aug = tf.while_loop(
+            lambda depth_level, image_aug: tf.less(depth_level, depth),
+            self._loop_on_depth,
+            [depth_level, image_aug],
+        )
+        result += tf.gather(chain_mixing_weights, curr_chain) * image_aug
+        curr_chain += 1
+        return image, chain_mixing_weights, curr_chain, result
+
+    def _auto_contrast(self, image):
+        return self.auto_contrast(image)
+
+    def _equalize(self, image):
+        return self.equalize(image)
+
+    def _posterize(self, image):
+        image = preprocessing.transform_value_range(
+            images=image,
+            original_range=self.value_range,
+            target_range=[0, 255],
+        )
+
+        bits = self._sample_level(self.severity, 3, tf.int32)
+        shift = tf.cast(4 - bits + 1, tf.uint8)
+        image = tf.cast(image, tf.uint8)
+        image = tf.bitwise.left_shift(tf.bitwise.right_shift(image, shift), shift)
+        image = tf.cast(image, self.compute_dtype)
+        return preprocessing.transform_value_range(
+            images=image,
+            original_range=[0, 255],
+            target_range=self.value_range,
+        )
+
+    def _rotate(self, image):
+        angle = tf.expand_dims(
+            self._sample_level(self.severity, 30, tf.float32), axis=0
```

Comment:
same comment as above, also should we expose maximum rotation value as a parameter?

Reply:
In the official implementation, these values are hardcoded. Do you think we should expose it as a parameter? What about the maximum values for other layers used in AugMix?

Reply:
Gotcha, good to know.  Let’s keep them hard coded unless we get a feature request.